node/src/reactor/participating/error.rs

```diff
@@ -39,4 +44,50 @@ pub enum Error {
     /// Failed to serialize data.
     #[error("serialization: {0}")]
     Serialization(#[source] bincode::ErrorKind),
+
+    /// Engine state error.
+    #[error("engine state: {0}")]
+    EngineState(#[from] engine_state::Error),
+
+    /// Block execution error.
+    #[error("block execution error: {0}")]
+    BlockExecutionError(#[from] BlockExecutionError),
+
+    /// [`bytesrepr`] error.
+    #[error("bytesrepr error: {0}")]
+    BytesReprError(#[from] bytesrepr::Error),
+
+    /// Cannot run genesis on pre-existing blockchain.
+    #[error("Cannot run genesis on pre-existing blockchain. First block: {first_block_header:?}")]
```

Comment:
```suggestion
    /// Cannot run genesis on preexisting blockchain.
    #[error("Cannot run genesis on preexisting blockchain. First block: {first_block_header:?}")]
```

(Or use "pre-existing" below, too.)